pilot/pkg/config/kube/ingress/conversion.go

```diff
@@ -308,7 +292,6 @@ func shouldProcessIngressWithClass(mesh *meshconfig.MeshConfig, ingress *v1beta1
 			return false
 		}
 	} else if ingressClass != nil {
-		// TODO support ingressclass.kubernetes.io/is-default-class annotation
```

Comment:
I think we still want this, see https://kubernetes.io/docs/concepts/services-networking/ingress/#default-ingress-class

Reply:
Took that out since when the `ingressclass.kubernetes.io/is-default-class` annotation is specified for an `IngressClass`, k8s automatically sets the `ingressClassName` field for Ingresses that don't specify it (weirdly didn't see that behavior documented on the k8s site), so our stuff should work automatically (since it will have `ingressClassName: istio` field already on it).

This does lead to confusing semantics around using both `IngressControllerMode` and the `ingressclass.kubernetes.io/is-default-class` annotation

Reply:
Oh nice, didn't know that! I would agree, IngressControllerMode was clearly designed before this new IngressClass stuff. but it seems not too bad, as the `is-default-class` will take precedence which I think is good.

STRICT+is-default-class = will be the default, as we see ingressClassName=istio
STRICT+another class is default = will ignore the default, as we see ingressClassName=not-istio, but handle the istio ones
DEFAULT+is-default-class =  will be the default, as we see ingressClassName=istio
DEFAULT+another class is default = will ignore it properly, as it will have ingressClassName=not-istio
DISABLED = will do nothing

so overall I think it makes sense, although a bit more complex than it should be